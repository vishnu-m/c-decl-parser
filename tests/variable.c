/* Declaration with storage class specifier */
static int list[20]; 

int *aptr[10]; // Declares an array of 10 pointers

/* Struct declaration */
struct st1 {
        int a;
};

/* Struct variable declaration and initialisation */
struct st1 st1_instance = {10};

/* Union declaration along with simultaneous union variable declaration */
union un1 {
        char *s;
}un1_instance;