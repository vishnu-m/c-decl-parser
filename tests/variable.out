VarDecl		"list"		storage class specifier = "static"		type = "int [20]"
VarDecl		"aptr"		storage class specifier = ""		type = "int *[10]"
StructDecl	"st1"		Parent = "variable.c"
FieldDecl	"a"		type = "int"		Parent = "st1"
VarDecl		"st1_instance"		storage class specifier = ""		type = "struct st1"
UnionDecl	"un1"		Parent = "variable.c"
FieldDecl	"s"		type = "char *"		Parent = "un1"
VarDecl		"un1_instance"		storage class specifier = ""		type = "union un1"
